### PR TITLE
It Starts

### PR DESCRIPTION
**Pull request by sherl0cks**
@mdanter @oybed @mcanoy 

See the DynamicPipelineJenkinsDSL which is the extension point into Jenkins. Everything after that does not have Jenkins dependencies. Worth noting too is the JettyTestServer, which is useful for working with a local Jenkins instance.

### CODE REVIEW

**Comment by sherl0cks** [comment]
Testing integration with slack

File: src/main/java/com/rhc/dynamic/pipeline/DynamicPipelineFactory.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	}
+
+	public DynamicPipelineFactory withConfigurationFile(String fileName) throws IOException {
+
+		InputStream is = getClass().getClassLoader().getResourceAsStream(fileName);
+		if (is == null) {
+			throw new RuntimeException("Could not find the specified configuration file: " + fileName);
+		}
+		this.configFile = IOUtils.toString(is);
+		LOGGER.debug(this.configFile);
```
Comment (mdanter) [inline src/main/java/com/rhc/dynamic/pipeline/DynamicPipelineFactory.java]: Please refactor all logging to use parametrized logging.
http://slf4j.org/faq.html#logging_performance

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		</repository>
+	</repositories>
+	<pluginRepositories>
+		<pluginRepository>
+			<id>repo.jenkins-ci.org</id>
+			<url>http://repo.jenkins-ci.org/public/</url>
+		</pluginRepository>
+	</pluginRepositories>
+	<properties>
+		<jenkins.version>1.642.3</jenkins.version>
```
Comment (sherl0cks) [inline pom.xml]: Take a look at the parent pom, which brings in a test suite for you. Also, https://wiki.jenkins-ci.org/display/JENKINS/Plugin+tutorial#Plugintutorial-PluginWorkspaceLayout

**Comment by sherl0cks** [comment]
Everything is cleaned up in latest commit

**Pull request merged**